pkg/rancher-desktop/main/mainEvents.ts

```diff
@@ -104,6 +105,15 @@ interface MainEventNames {
    */
   'diagnostics-trigger'(id: string): void;
 
+  /**
+   * Force trigger diagnostics with the given id.
+   * This is used when something has changed that might affect whether the given
+   * diagnostic needs to be re-run.
+   * @note This does not update the last run time (since it only runs a single
+   * checker).
+   */
+  'diagnostics-trigger'(id: string): DiagnosticsCheckerResult | undefined;
```

Comment:
There's already an `diagnostics-trigger` event (just above); duplicating this appears to be confusing eslint. The net effect is that it shadows the previous method, and we _only_ have an even handler (instead of listener channel) instead.

Reply:
I opted to remove the listener in favor of handler for this use case.